pkg/scalers/cron_scaler.go

```diff
@@ -0,0 +1,166 @@
+package scalers
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/robfig/cron/v3"
+	"k8s.io/api/autoscaling/v2beta1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	defaultDesiredReplicas = 1
+	cronMetricName         = "ReplicaCount"
+	cronMetricType         = "External"
+)
+
+type cronScaler struct {
+	metadata               *cronMetadata
+}
+
+type cronMetadata struct {
+	start            string
+	end              string
+	timezone         string
+	desiredReplicas  int64
+}
+
+var cronLog = logf.Log.WithName("cron_scaler")
+
+// NewCronScaler creates a new cronScaler
+func NewCronScaler(resolvedEnv, metadata map[string]string) (Scaler, error) {
+	meta, parseErr := parseCronMetadata(metadata, resolvedEnv)
+	if parseErr != nil {
+		return nil, fmt.Errorf("error parsing cron metadata: %s", parseErr)
+	}
+
+	return &cronScaler{
+		metadata               : meta,
```

Comment:
Is the formatting here correct?

Reply:
have corrected the format

Reply:
great, Just checking whether you have formatted all the files (including tests)? (`go fmt` or use the one in the IDE)

Reply:
@zroubalik done. I have run gofmt now. Please check

Reply:
@ckuduvalli thanks! looking good